lib/ASTGen/Sources/ASTGen/Patterns.swift

```diff
@@ -45,6 +50,60 @@ extension ASTGenVisitor {
       loc: nameLoc
     )
   }
+
+  func generate(isTypePattern node: IsTypePatternSyntax) -> BridgedIsPattern {
+    return .createParsed(
+      self.ctx,
+      isLoc: self.generateSourceLoc(node.isKeyword),
+      typeExpr: .createParsed(
+        self.ctx,
+        type: self.generate(type: node.type)
+      )
+    )
+  }
+
+  func generate(missingPattern node: MissingPatternSyntax) -> BridgedPattern {
+    fatalError("unimplemented")
+  }
+
+  func generate(tuplePattern node: TuplePatternSyntax) -> BridgedPattern {
+    if node.elements.count == 1 && node.elements.first!.label == nil {
```

Comment:
```suggestion
    if node.elements.count == 1, let firstElement = node.elements.first, firstElement.label == nil {
```

And then `firstElement` can be re-used further down (`firstElement.pattern`).